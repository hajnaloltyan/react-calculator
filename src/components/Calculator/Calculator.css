.calculator {
  display: flex;
  flex-direction: column;
}

.result {
  grid-column: 1 / -1;
  background-color: #929090;
  padding: 10px;
  text-align: right;
  color: #fff;
}
