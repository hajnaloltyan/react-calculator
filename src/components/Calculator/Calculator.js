import React, { useState } from 'react';
import './Calculator.css';
import Buttons from '../Buttons/Buttons';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newCalculatorData = calculate(calculatorData, buttonName);
    setCalculatorData(newCalculatorData);
  };

  const displayResult = calculatorData.next || calculatorData.total || '0';

  return (
    <section className="calculator">
      <h2 className="result" data-testid="result">{displayResult}</h2>
      <Buttons onButtonClick={handleButtonClick} />
    </section>
  );
};

export default Calculator;
