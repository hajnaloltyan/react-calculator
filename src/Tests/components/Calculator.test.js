import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../../components/Calculator/Calculator';

describe('Calculator', () => {
  test('component to be present and rendered', () => {
    const test = render(<Calculator />);
    expect(test).toMatchSnapshot();
  });

  test('component to have result showing 0', () => {
    const { getByTestId } = render(<Calculator />);
    const resultValue = Number(getByTestId('result').textContent);
    expect(resultValue).toEqual(0);
  });
});
