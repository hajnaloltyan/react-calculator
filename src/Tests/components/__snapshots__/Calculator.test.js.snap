// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component to be present and rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="calculator"
      >
        <h2
          class="result"
          data-testid="result"
        >
          0
        </h2>
        <article
          class="buttons"
        >
          <button
            class="btn-text"
            type="button"
          >
            AC
          </button>
          <button
            class="btn-text"
            type="button"
          >
            +/-
          </button>
          <button
            class="btn-text"
            type="button"
          >
            %
          </button>
          <button
            class="btn-text orange"
            type="button"
          >
            รท
          </button>
          <button
            class="btn-text"
            type="button"
          >
            7
          </button>
          <button
            class="btn-text"
            type="button"
          >
            8
          </button>
          <button
            class="btn-text"
            type="button"
          >
            9
          </button>
          <button
            class="btn-text orange"
            type="button"
          >
            x
          </button>
          <button
            class="btn-text"
            type="button"
          >
            4
          </button>
          <button
            class="btn-text"
            type="button"
          >
            5
          </button>
          <button
            class="btn-text"
            type="button"
          >
            6
          </button>
          <button
            class="btn-text orange"
            type="button"
          >
            -
          </button>
          <button
            class="btn-text"
            type="button"
          >
            1
          </button>
          <button
            class="btn-text"
            type="button"
          >
            2
          </button>
          <button
            class="btn-text"
            type="button"
          >
            3
          </button>
          <button
            class="btn-text orange"
            type="button"
          >
            +
          </button>
          <button
            class="btn-text btn-0"
            type="button"
          >
            0
          </button>
          <button
            class="btn-text"
            type="button"
          >
            .
          </button>
          <button
            class="btn-text orange"
            type="button"
          >
            =
          </button>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="calculator"
    >
      <h2
        class="result"
        data-testid="result"
      >
        0
      </h2>
      <article
        class="buttons"
      >
        <button
          class="btn-text"
          type="button"
        >
          AC
        </button>
        <button
          class="btn-text"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn-text"
          type="button"
        >
          %
        </button>
        <button
          class="btn-text orange"
          type="button"
        >
          รท
        </button>
        <button
          class="btn-text"
          type="button"
        >
          7
        </button>
        <button
          class="btn-text"
          type="button"
        >
          8
        </button>
        <button
          class="btn-text"
          type="button"
        >
          9
        </button>
        <button
          class="btn-text orange"
          type="button"
        >
          x
        </button>
        <button
          class="btn-text"
          type="button"
        >
          4
        </button>
        <button
          class="btn-text"
          type="button"
        >
          5
        </button>
        <button
          class="btn-text"
          type="button"
        >
          6
        </button>
        <button
          class="btn-text orange"
          type="button"
        >
          -
        </button>
        <button
          class="btn-text"
          type="button"
        >
          1
        </button>
        <button
          class="btn-text"
          type="button"
        >
          2
        </button>
        <button
          class="btn-text"
          type="button"
        >
          3
        </button>
        <button
          class="btn-text orange"
          type="button"
        >
          +
        </button>
        <button
          class="btn-text btn-0"
          type="button"
        >
          0
        </button>
        <button
          class="btn-text"
          type="button"
        >
          .
        </button>
        <button
          class="btn-text orange"
          type="button"
        >
          =
        </button>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
